state/remoteapplication.go

```diff
@@ -29,12 +29,13 @@ type RemoteApplication struct {
 
 // remoteApplicationDoc represents the internal state of a remote application in MongoDB.
 type remoteApplicationDoc struct {
-	DocID         string              `bson:"_id"`
-	Name          string              `bson:"name"`
-	URL           string              `bson:"url"`
-	Endpoints     []remoteEndpointDoc `bson:"endpoints"`
-	Life          Life                `bson:"life"`
-	RelationCount int                 `bson:"relationcount"`
+	DocID           string              `bson:"_id"`
```

Comment:
The vast majority of our mongo structs do not have fields with doc - these structs exist only to get stuff into /out of the db. The business object layer is however documented, see the methods on RemoteApplication. What's done here is following Juju's standard practice.